ansible_collections/arista/avd/roles/eos_designs/python_modules/underlay/utils.py

```diff
@@ -184,3 +188,108 @@ def _get_l3_interface_cfg(self, l3_interface: dict) -> dict | None:
             interface["flow_tracker"] = {"hardware": self.shared_utils.wan_flow_tracker_name}
 
         return strip_empties_from_dict(interface)
+
+    def _get_l3_uplink_with_l2_as_subint(self, link: dict) -> (dict, list[dict]):
+        """
+        Return a tuple with main uplink interface, list of subinterfaces representing each SVI.
+        """
+        vlans = [int(vlan) for vlan in range_expand(link["vlans"])]
+
+        # Main interface
+        # Routed interface with no config unless there is an SVI matching the native-vlan, then it will contain the config for that SVI
+
+        interfaces = []
+        for tenant in self.shared_utils.filtered_tenants:
+            for vrf in tenant["vrfs"]:
+                for svi in vrf["svis"]:
+                    svi_id = int(svi["id"])
+                    # Skip any vlans not part of the link
+                    if svi_id not in vlans:
+                        continue
+
+                    interfaces.append(self._get_l2_as_subint(link, svi, vrf))
+
+        # If we have the main interface covered, we can just exclude it from the list and return as main interface.
+        # Otherwise we return an almost empty dict as the main interface since it was already covered by the calling function.
+        main_interface = get_item(interfaces, "name", link["interface"], default={"type": "routed", "mtu": self.shared_utils.p2p_uplinks_mtu})
+        main_interface.pop("description", None)
```

Comment:
I see why we are popping the description but would it make sense to have the main interface description prefixes to the SVI descriptions ?